### PR TITLE
Update to be compliant with SMOD 3.6.0

### PR DESCRIPTION
**Pull request by joker-119**
To prevent server crashes when firing the OnDisconnect event (side note, this event never had access to the IP addresses KARL. :P)

Unfortunately, no Connection.IsBanned in OnPlayerLeave, and no access to the player object. I'm sure you can fix this, but I don't wanna jank it and upset you c:

### CODE REVIEW

**Comment by KarlOfDuty** [comment]
I don't see any of these changes on the Smod repo.

**Comment by joker-119** [comment]
They are all detailed in the #tester-requests versions from Mith, most notably version 3.5.1-D which deprecates OnLateDisconnect and OnDisconnect in favor of the new OnPlayerLeave, i believe 3.5.1-C changed the handcuff event's .Handcuffed bool to .Allow.

The reason for me making this PR now as opposed to when 3.5.1 releases, is because hooking the OnDisconnect and OnLateDisconnect event has been crashing my server on 3.5.1-D (using -C before these changes took place is perfectly stable).

**Comment by KarlOfDuty** [comment]
Well, that is why they are tester versions. I don't think it's a good idea to add this before the release, especially before they have even been added to the official repo.

File: SCPDiscordPlugin/Config.cs [hunk 1]
```diff
@@ -45,8 +45,9 @@ public static class Config
             // Round events
             { "channels.onroundstart",              new string[]{ } },
             { "channels.onconnect",                 new string[]{ } },
-            { "channels.ondisconnect.default",      new string[]{ } },
-            { "channels.ondisconnect.banned",       new string[]{ } },
+          //{ "channels.ondisconnect.default",      new string[]{ } },
+          //{ "channels.ondisconnect.banned",       new string[]{ } },
```
Comment (KarlOfDuty) [inline SCPDiscordPlugin/Config.cs]: If these do not exist anymore they should probably be removed.

File: SCPDiscordPlugin/EventListeners/RoundEventListener.cs [hunk 2]
```diff
@@ -36,15 +36,18 @@ public void OnConnect(ConnectEvent ev)
             this.plugin.SendMessage(Config.GetArray("channels.onconnect"), "round.onconnect", variables);
         }
 
+        /*
         /// <summary>
         ///  This is the event handler for disconnection events.
         /// </summary>
+        **DEPRICATED**
```
Comment (KarlOfDuty) [inline SCPDiscordPlugin/EventListeners/RoundEventListener.cs]: Same with this, no need to keep a bunch of commented out code around.

Comment (joker-119) [reply SCPDiscordPlugin/EventListeners/RoundEventListener.cs]: Yea, I wasn't sure what would/wouldn't change between 3.5.1 testing and 3.6.0 so I kept them in just incase some changes got reverted

Comment (KarlOfDuty) [reply SCPDiscordPlugin/EventListeners/RoundEventListener.cs]: It would seem the OnDisconnect event is not getting removed so I readded that.

**Review by KarlOfDuty — APPROVED** [review]

**Pull request merged**